contracts/vcon/ERC20Lockable.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "solmate/tokens/ERC20.sol";
+
+/**
+ @title ERC20 Lockable contract
+ @author eswak
+ @notice an ERC20 extension which allows users to define a locking address.
+ When a user's tokens are locked in an address, they can't be transferred
+ anywhere else.
+ */
+abstract contract ERC20Lockable is ERC20 {
+    /// @notice thrown when violating the lock logic
+    error LockError();
+
+    /// @notice mapping of users to lock address
+    mapping(address => address) public lockAddress;
+
+    /// @dev Emitted when a `user` lock their tokens on `lock`.
+    event Lock(address indexed user, address indexed lock);
+
+    /// @dev Emitted when a `user` unlock their tokens from `lock`.
+    event Unlock(address indexed user, address indexed lock);
+
+    /// @notice set the locking address of an account
+    function _setLockAddress(address user, address userLock) internal {
+        if (userLock == address(0)) revert LockError(); // cannot lock to address(0)
+        if (lockAddress[user] != address(0)) revert LockError(); // already locked
+
+        lockAddress[user] = userLock;
+
+        emit Lock(user, userLock);
+
+        // allow locker to pull user's tokens
+        allowance[user][userLock] = type(uint256).max;
+        emit Approval(user, userLock, type(uint256).max);
+    }
+
+    /// @notice unset the locking address of an account
+    function _unsetLockAddress(address user) internal {
+        address userLock = lockAddress[user];
+        if (userLock == address(0)) revert LockError(); // not locked
+
+        delete lockAddress[user];
+
+        emit Unlock(user, userLock);
+
+        // reset locker's allowance on user's tokens
+        allowance[user][userLock] = 0;
+        emit Approval(user, userLock, 0);
+    }
+
+    /// @notice lock msg.sender's tokens on a locker
+    function lock(address locker) external {
+        _setLockAddress(msg.sender, locker);
+    }
+
+    /// @notice unlock a user that is locking on msg.sender
+    function unlock(address user) external {
+        if (msg.sender != lockAddress[user]) revert LockError(); // invalid unlocker
+        _unsetLockAddress(user);
+    }
```

Comment:
we need another feature that stops users from transferring tokens if the global transfer ability is paused

Reply:
yep I've put this logic in VconV2 directly